mailman3_fedmsg_plugin.py

```diff
@@ -67,47 +78,44 @@ class Archiver(object):
         "user-agent",
     ]
 
-    def __init__(self):
-        if not getattr(getattr(fedmsg, '__local', None), '__context', None):
-            hostname = socket.gethostname().split('.')[0]
-            fedmsg.init(name="mailman.%s" % hostname)
-        self.config = fedmsg.config.load_config()
-
-
     def archive_message(self, mlist, msg):
-        """Send the message to the "archiver".
+        """
+        Send the message to the "archiver".
 
         In our case, we just publish it to the fedmsg bus.
 
         :param mlist: The IMailingList object.
         :param msg: The message object.
         """
 
-        if mlist.list_name in self.config.get('mailman.excluded_lists', []):
+        if mlist.list_name in conf["consumer_config"].get("excluded_lists", []):
             return
 
-        # Here, by `str` we mean `unicode`.  We're python3 only!
-        format = lambda value: value and str(value)
-        msg_metadata = dict([(k, format(msg.get(k))) for k in self.keys])
-        lst_metadata = dict(
-            list_name=mlist.list_name,
-            mail_host=mlist.mail_host,
-            fqdn_listname=mlist.fqdn_listname,
-            list_id=mlist.list_id,
-            display_name=mlist.display_name,
-            )
-
-        fedmsg.publish(topic='receive', modname='mailman',
-                       msg=dict(msg=msg_metadata, mlist=lst_metadata))
+        def _format(value):
+            return str(value) if value else value
+
+        msg_metadata = {key: _format(getattr(msg, key, None)) for key in self.keys}
+        lst_metadata = {
+            "list_name": mlist.list_name,
+            "mail_host": mlist.mail_host,
+            "fqdn_listname": mlist.fqdn_listname,
+            "list_id": mlist.list_id,
+            "display_name": mlist.display_name,
+        }
+        body = dict(msg=msg_metadata, mlist=lst_metadata)
+        fm_msg = Message(body=body, topic="mailman.receive")
+        safe_publish(fm_msg)
 
     def list_url(self, mlist):
-        """ This doesn't make sense for fedmsg.
+        """
+        This doesn't make sense for fedmsg.
         But we must implement for IArchiver.
         """
         return None
 
     def permalink(self, mlist, msg):
```

Comment:
Method definition identical to https://github.com/fedora-infra/mailman3-fedmsg-plugin/blob/fedora-messaging/mailman3_fedmsg_plugin/__init__.py#L95